### ISSUE TITLE
Domain WG Applicant: Cryptography

### ISSUE THREAD

**Issue by skade** [rust-lang/wg-governance#12]
This issue acts as a tracking issue and log for the application of a cryptography domain WG.

Details can be found here:
https://internals.rust-lang.org/t/enabling-the-formation-of-new-working-groups/10218/12

Current Status:
* [x] Application sent
* [ ] Charta written

The initial goal would be to find a pair of governance WG members to facilitate the formation of the group from our side.

**Comment by skade** [comment]
Personal opinion: Cryptography in Rust is a topic with some sharp edges, but also quite a lot of opportunity around safety. I'd be very glad if time were spent to explore practices here.

**Comment by tarcieri** [comment]
Looks like the next steps here are:

1. Recruit interested people to help run the WG
2. Write charter

**Comment by briansmith** [comment]
I'm also interested in leading this working group.

**Comment by rozbb** [comment]
I would also like to help run this

**Comment by tarcieri** [comment]
@newpavlov @eternaleye is this something either of you are interested in helping organize / participating in?

**Comment by newpavlov** [comment]
I am definitely interested in participating! Though I am not sure if I will have enough time in near future to help with organizing WG or drafting its documents, but I will be happy to participate in WG discussions and will try to keep my finger on the pulse!

**Comment by nickray** [comment]
I'm very interested in participating in this WG, from the point of view of [my Zissou project](https://github.com/nickray/zissou): Implementing a PIV-like on Cortex-M4/33 in USB stick form factor (parts 2+3 of the PIV spec, but with part 1 swapped out to enable newer cryptography).

**Comment by tarcieri** [comment]
@nickray neat! I imagine there are many ways the Embedded WG and Cryptography WG could collaborate.

**Comment by burdges** [comment]
I'm interested, limited time, but hey.  

I suppose the largest questions is:  Who would drive any changes around LLVM's code generation?  We still have issues there if I recall.  And a rust compiler working group would largely be worried about those issues, right?   

Zeroing memory efficiently sounds like an issue that also does not completely avoid rustc and LLVM, like maybe intrinsics for stack instrumentation.

**Comment by str4d** [comment]
I'm interested in participating in this WG.

**Comment by tarcieri** [comment]
> I suppose the largest questions is: Who would drive any changes around LLVM's code generation?

I think that's a bit ambitious for an initial work item. My 2c

**Comment by eternaleye** [comment]
@tarcieri I'm definitely interested

**Comment by tarcieri** [comment]
Sweet, well that's 8 people including myself which seems like a good start.

@skade is the next step on our end writing a charter, or should we wait until there are governance WG liasons available?

**Comment by tarcieri** [comment]
@KizzyCode is this something you'd be interested in participating in?

**Comment by KizzyCode** [comment]
@tarcieri Yeah, why not? I'm currently busy studying and it would be my first WG so I don't really know if I'm up to this task – but I'll be happy to contribute where I can 😄

**Comment by NikVolf** [comment]
Hey, would be great to participate, I worked with crypto in Rust since there were no crypto in Rust :)

**Comment by erasmospunk** [comment]
Interested in participating. I have a several year experience in developing a cryptocurrency wallet and contributing to projects in the space.

**Comment by phayes** [comment]
I would be interested in participating in this working group as well.

**Comment by taralx** [comment]
Here from security wg - would also be willing to contribute.

**Comment by isislovecruft** [comment]
I'm also willing to participate.

**Comment by mimoo** [comment]
I'm late, but I'm interested

**Comment by WildCryptoFox** [comment]
Huh, I thought I already said this - I'm definitely interested!

**Comment by KizzyCode** [comment]
Maybe this is a stupid question, but have there been any activities yet? 😅

**Comment by tarcieri** [comment]
Nope. The next step is to write a charter, which is something I’m planning to post a draft of here soon.

**Comment by tarcieri** [comment]
Here is a draft charter, based on some of the things I [posted on the original rust-internals thread](https://internals.rust-lang.org/t/enabling-the-formation-of-new-working-groups/10218/12):

-- snip --

Our mission is to foster a robust Rust cryptography ecosystem.

- Provide overall direction for cryptography in Rust through the creation of an interoperable cryptographic ecosystem by producing core cryptography crates
- Act as a resource for the other teams/WGs when they have questions about the use of cryptography in Rust projects
- Create an interest group and focal point for Rust cryptography in general

**Comment by brycx** [comment]
Very late to the party, but I'm also interested in participating and contributing with whatever I can.

Are there any news on this WG? If not, what are the next steps?

**Comment by tarcieri** [comment]
As far as I know we're waiting for a response from the core team

**Comment by huitseeker** [comment]
I'm keen to help out this WG as well.

**Comment by michalfita** [comment]
Is there any particular reason why this sits quiet?

**Comment by Manishearth** [comment]
We're still evaluating the utility of having "official" domain WGs.

**Comment by tarcieri** [comment]
In the past there's been some discussion of "interest groups" which are largely discussion oriented instead of officially blessed working groups.

I think "cryptography" is a great use case for exploring Rust "interest groups". As a minimum ask: how about creating an `#ig-cryptography` on the official Rust Zulip and/or Discord?

**Comment by tarcieri** [comment]
FYI, the @RustCrypto project ended up making our own Zulip:

https://rustcrypto.zulipchat.com/

I think it worked out pretty well as we ended up with quite a few streams since there are a lot of cryptographic domains we have crates for. I invite everyone here to join!

**Comment by tarcieri** [comment]
For people specifically interested in a "Cryptography Interest Group" as discussed above, we've made an unofficial Zulip stream specific to that purpose: https://rustcrypto.zulipchat.com/#narrow/stream/262272-cryptography-ig/topic/welcome

**Comment by tarcieri** [comment]
For anyone who missed it, we just had a kickoff call for a Rust Cryptography Interest Group:

https://www.youtube.com/watch?v=q6bEfN8v-98

We're presently trying to decide on other communication channels, including what chat and AV platforms to use going forward, if anyone is interested:

https://github.com/The-DevX-Initiative/RCIG_Coordination_Repo/issues/1
